cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -84,8 +84,11 @@ public TaskWorkerHttpHandlerInternal(CConfiguration cConf, Consumer<String> stop
     this.runnableTaskLauncher = new RunnableTaskLauncher(cConf);
     this.metadataServiceEndpoint = cConf.get(Constants.TaskWorker.METADATA_SERVICE_END_POINT);
     this.stopper = s -> {
-      stopper.accept(s);
-      inflightRequests.decrementAndGet();
+      if (cConf.getBoolean(Constants.TaskWorker.CONTAINER_KILL_AFTER_EXECUTION)) {
+        stopper.accept(s);
+      } else {
+        inflightRequests.decrementAndGet();
```

Comment:
There is a catch block for ClassNotFoundException and ClassCastException in /task and /run end points. The stopper is not called then , but inflightRequests seems like should decrement for that case also?

Reply:
this.stopper consumer in the constructor get initialized to both stopping the service and also taking care of inflight requests.

Reply:
I was referring to https://github.com/cdapio/cdap/blob/356a15d7376e2b49c27c91e1caef8e9906b49b25/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java#L114 where the stopper.accept() is not invoked

Reply:
Thanks. Fixed now.